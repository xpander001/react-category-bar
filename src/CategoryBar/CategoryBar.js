import React, { Component } from 'react';

import './CategoryBar.scss';

class CategoryBar extends Component {
  
  constructor(props) {
    
    super(props);

    this.state = {
      isDropdownOpen: false,
      selectedId: props.categories[0].id
    }
  }

  selectCategory(selectedCategory) {
    this.setState({selectedId: selectedCategory.id});
    this.props.onCategorySelected(selectedCategory);
  }

  // Lifecycle methods
  componentDidMount() {

  }

  getDropdown() {
    if (this.state.isDropdownOpen) {
      return (
        <div className="CategoryBar__Dropdown">
          <span className="CategoryBar__Dropdown__Title"></span>
          <ul className="CategoryBar__Dropdown__List">
          </ul>
        </div>
      );
    }
    return null;
  }
  
  render() {
    return (
      <div className="CategoryBar">
        <ul className="CategoryBar__List">
          {
            this.props.categories.map((category) => 
              <li className={
                this.state.selectedId === category.id ? 
                "CategoryBar__Category CategoryBar__Category--Selected" :
                "CategoryBar__Category"}
                onClick={() => {
                  this.selectCategory(category);
                }}
                key={category.id}>
                
                {category.name}
              </li>
            )
          }
        </ul>
        {this.getDropdown()}
      </div>
    );
  }
}

export default CategoryBar;