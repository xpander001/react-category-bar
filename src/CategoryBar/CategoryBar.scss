// MIXINGS

.CategoryElement {
  color: black;
  cursor: pointer;
  display: inline-block;
  font-size: 18px;
  font-weight: 600;
  letter-spacing: .9px;
  line-height: normal;
  margin-right: 0px;
  padding: 0 30px 10px 30px;
  position: relative;
  text-transform: capitalize;
  vertical-align: top;
  text-align: center;
  border-bottom: 2px solid white;
  -webkit-font-smoothing: subpixel-antialiased;
  height: 34px;
  &:hover {
    color: grey;
  }
}

.CategoryElementSelected {
  color: blue;
  border-bottom: 2px solid currentColor;
  font-weight: bold;
}

// BEM for CategoryBar

.CategoryBar {
  display: flex;
  font-size: 0;
	width: auto;
	height: 34px;
  clear: both;
  overflow: visible;

  &__List {
    flex: 1;
    list-style: none;
    margin: 0px;
    padding: 0px;
    height: 34px;
    overflow: hidden;

    &__Category {
      @extend .CategoryElement;

      &--Selected {
        @extend .CategoryElementSelected;
      }
    }

  }

}