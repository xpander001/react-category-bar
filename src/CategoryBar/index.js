import CategoryBar from './CategoryBar';

export default CategoryBar;