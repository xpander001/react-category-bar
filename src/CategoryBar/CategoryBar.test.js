import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CategoryBar from './CategoryBar';


describe('SearchBar', () => {

  const categories = [ 
    {
      id: 1,
      name: 'Category 1'
    },
    {
      id: 2,
      name: 'Category 2',
    },
    {
      id: 3,
      name: 'Category 3',
    },
    {
      id: 4,
      name: 'Category 4',
    },
    {
      id: 5,
      name: 'Category 5',
    },
    {
      id: 6,
      name: 'Category 6',
    },
    {
      id: 7,
      name: 'Category 7',
    },
    {
      id: 8,
      name: 'Category 8',
    }
  ];

  let onCategorySelected;

  beforeEach(() => {
    onCategorySelected = jest.fn();
  });

  test('renders without crashing', () => {
    const component = shallow(<CategoryBar categories={categories} onCategorySelected={onCategorySelected}/>);
    expect(toJson(component)).toMatchSnapshot();
  });

  test('should have first category selected by defaul', () => {
    const component = shallow(<CategoryBar categories={categories} onCategorySelected={onCategorySelected}/>);
    expect(component.state().selectedId).toBe(categories[0].id);
  });

  test('should mark category as selected when selecting category', () => {
    const component = shallow(<CategoryBar categories={categories} onCategorySelected={onCategorySelected}/>);
    const categoriesList = component.find('.CategoryBar__List');
    let thirdCategory = categoriesList.childAt(2);
    thirdCategory.simulate('click');
    expect(component.state().selectedId).toBe(categories[2].id);
  });
  
  test('should call onCategorySelected property when category clicked', () => {
    const component = shallow(<CategoryBar categories={categories} onCategorySelected={onCategorySelected}/>);
    const categoriesList = component.find('.CategoryBar__List');
    let thirdCategory = categoriesList.childAt(2);
    thirdCategory.simulate('click');
    expect(onCategorySelected.mock.calls.length).toBe(1);
    expect(onCategorySelected.mock.calls[0]).toEqual([categories[2]]);
  });
  
  
})
