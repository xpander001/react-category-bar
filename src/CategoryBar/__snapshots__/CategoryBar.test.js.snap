// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders without crashing 1`] = `
<div
  className="CategoryBar"
>
  <ul
    className="CategoryBar__List"
  >
    <li
      className="CategoryBar__List__Category CategoryBar__List__Category--Selected"
      onClick={[Function]}
    >
      Category 1
    </li>
    <li
      className="CategoryBar__List__Category"
      onClick={[Function]}
    >
      Category 2
    </li>
    <li
      className="CategoryBar__List__Category"
      onClick={[Function]}
    >
      Category 3
    </li>
    <li
      className="CategoryBar__List__Category"
      onClick={[Function]}
    >
      Category 4
    </li>
    <li
      className="CategoryBar__List__Category"
      onClick={[Function]}
    >
      Category 5
    </li>
    <li
      className="CategoryBar__List__Category"
      onClick={[Function]}
    >
      Category 6
    </li>
    <li
      className="CategoryBar__List__Category"
      onClick={[Function]}
    >
      Category 7
    </li>
    <li
      className="CategoryBar__List__Category"
      onClick={[Function]}
    >
      Category 8
    </li>
  </ul>
</div>
`;
